src/Featurizers/LagLeadOperatorFeaturizer.h

```diff
@@ -0,0 +1,412 @@
+                for (auto it = offsets.begin(); it != offsets.end(); it++) {
+                    *it -= min;
+                }
+                return offsets;
+            }()
+        )
+    ),
+    // if there's only one offset, we use the absolute value of it plus horizon as the size
+    // otherwise we need to check if the minimum offset is negative:
+    // if it is, the buffer size is the absolute value of minimum offset plus horizon
+    // if it's not, the buffer size is the difference between min and max of offsets, which is already calculated in the offsets calibration so we can use max_element directly
+    _buffer(
+        _horizon + 
+        (_offsets.size() == 1 ? 
+        static_cast<size_t>(std::abs(_offsets[0] + _offset_delta)) : 
+        static_cast<size_t>(
+            *std::max_element(_offsets.cbegin(), _offsets.cend()) + _offset_delta < 0 ? 
+            std::abs(_offset_delta) : 
+            *std::max_element(_offsets.cbegin(), _offsets.cend())))
+    ) {
+        // prepopulate circular buffer with null values to imitate non-existing prior rows
+        for (std::int64_t i = 0; i < (static_cast<std::int64_t>(_horizon) - _offset_delta - 1); ++i) {
+            _buffer.push(Traits<TransformedTargetType>::CreateNullValue());
+        }
+}
+
+template <typename T>
+LagLeadOperatorTransformer<T>::LagLeadOperatorTransformer(Archive &ar) :
+    LagLeadOperatorTransformer(
+        [&ar](void) {
+            // Version
+            std::uint16_t                   majorVersion(Traits<std::uint16_t>::deserialize(ar));
+            std::uint16_t                   minorVersion(Traits<std::uint16_t>::deserialize(ar));
+
+            if(majorVersion != 1 || minorVersion != 0)
+                throw std::runtime_error("Unsupported archive version");
+
+            // Data
+            std::uint32_t horizon = Traits<std::uint32_t>::deserialize(ar);
+            std::vector<std::int64_t> offsets = Traits<std::vector<std::int64_t>>::deserialize(ar);
+            
+            return LagLeadOperatorTransformer(std::move(horizon), std::move(offsets));
+        }()
+    ) {
+}
+
+template <typename T>
+bool LagLeadOperatorTransformer<T>::operator==(LagLeadOperatorTransformer const &other) const {
+    return this->_horizon == other._horizon && this->_offset_delta == other._offset_delta
+            && this->_offsets == other._offsets;
+}
+
+template <typename T>
+bool LagLeadOperatorTransformer<T>::operator!=(LagLeadOperatorTransformer const &other) const {
+    return (*this == other) == false;
+}
+
+template <typename T>
+void LagLeadOperatorTransformer<T>::save(Archive &ar) const /*override*/ {
+    // Version
+    Traits<std::uint16_t>::serialize(ar, 1); // Major
+    Traits<std::uint16_t>::serialize(ar, 0); // Minor
+
+    // Data
+    Traits<std::uint32_t>::serialize(ar, _horizon);
+
+    // reset _offsets to its input version
+    std::vector<std::int64_t> offsets(_offsets);
```

Comment:
>offsets [](start = 30, length = 7)

deltas

Reply:
I'm not sure I understand this comment.

Reply:
The concept of deltas (signed values) and offsets (unsigned values) are being mixed up. In this case, you are serializing deltas rather than offsets.

Reply:
Resolved per offline discussion. Changed names according to offset and delta concepts